File: app/src/main/java/com/caitlynwiley/pettracker/activities/BaseActivity.java
package com.caitlynwiley.pettracker.activities;

import android.os.Bundle;
import android.preference.PreferenceManager;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.caitlynwiley.pettracker.R;

public class BaseActivity extends AppCompatActivity {

    private boolean useDarkTheme;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        useDarkTheme = PreferenceManager.getDefaultSharedPreferences(this).getBoolean("dark_theme_enabled", false);
        setTheme(useDarkTheme ? R.style.DarkTheme : R.style.LightTheme);
        super.onCreate(savedInstanceState);
    }

    @Override
    protected void onResume() {
        boolean darkThemePref = PreferenceManager.getDefaultSharedPreferences(this).getBoolean("dark_theme_enabled", false);
        setTheme(useDarkTheme ? R.style.DarkTheme : R.style.LightTheme);
        if (useDarkTheme != darkThemePref) {
            useDarkTheme = darkThemePref;
            this.recreate();
        }
        super.onResume();
    }
}

File: app/src/main/java/com/caitlynwiley/pettracker/activities/BaseActivity.kt
package com.caitlynwiley.pettracker.activities

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.viewModels

import androidx.compose.animation.ExperimentalAnimationApi
import androidx.compose.runtime.livedata.observeAsState
import androidx.compose.ui.ExperimentalComposeUiApi
import com.caitlynwiley.pettracker.theme.PetTrackerTheme
import com.caitlynwiley.pettracker.viewmodel.AppWideViewModel

open class BaseActivity : ComponentActivity() {
    private val appViewModel by viewModels<AppWideViewModel>()

    @ExperimentalComposeUiApi
    @ExperimentalAnimationApi
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContent {
            val user = appViewModel.user.observeAsState()
            PetTrackerTheme {
                if (user.value == null) {
                    LoginActivity()
                } else {
                    MainActivity()
                }
            }
        }
    }
}